### PR TITLE
Limit non-English locales

### LINKED ISSUE #3665: Server start error due to _githistory.json [retired languages]
Following the deletion of the locales that were moved to a new repository, the Yari server throws an error for these locales when trying to launch it because it cannot find the `_githistory.json`.

To fix the server on my side, I reset my main branch before the deletion and restarted the server to at least have the `_githistory.json` once and not have any more problems, but users who will create the server with `yari`, `content` and `translated-content` might get an error to tell them that `/ar/_githistory.json` doesn't exist and the same for the other locales deleted from `translated-content`.

And on the Window side, nothing worked: `yarn build` returns an error, `yarn dev` returns an error. For the moment I have to keep the `_githistory.json` to be able to launch the server.

Maybe something needs to change on the Yari side so that withdrawn locals are not caught looking for them?

### PR DESCRIPTION
**Pull request by escattone**
Fixes #3665 

Part of #3642 

**TODO (after merge)**:
- [ ] Update `*/docs/*` Cloudfront behavior to use `MDN-None` forwarding policy on prod & dev (stage as already been done)
- [ ] Update the version of the content-origin-request lambda@edge function in our dev, stage, and prod Cloudfront behaviors

### CODE REVIEW

**Comment by escattone** [comment]
@peterbe @Gregoor I don't understand why these tests are failing.

**Comment by escattone** [comment]
@peterbe @Gregoor I just discovered one thing I need to do is remove any URL's with retired locales from the `_redirects.txt` files in content and translated-content.

**Comment by escattone** [comment]
> @peterbe @Gregoor I don't understand why these tests are failing.

@peterbe @Gregoor This was due to the presence of a URL with a retired locale in the content repo's `files/en-us/_redirects.txt`. All good now (see https://github.com/mdn/content/pull/4627). This ready for review.

**Comment by escattone** [comment]
@pertebe @Gregoor FYI, at first I implemented the retired-locale redirects directly in the lambda function, but then thought that perhaps a better place for that functionality was in the fundamental redirects (e.g., links to retired locales would be detected as flaws). As it stand, this PR places the redirect code in the fundamental redirects, but if you simply remove the [last commit](https://github.com/mdn/yari/pull/3661/commits/6c39c1ffc61e38e807645f80e7e84571f40e4357), the placement moves back into the lambda function. So we can easily go either way, depending on what we all think is best.

**Comment by Gregoor** [comment]
Using the flaws sounds like a great idea to me and the codes does cohere to me. I am a little sad that we don't have code-sharing between the client and lib's constants, but it's only incidentally the first time I am noticing it and not too related to this PR (plus you kept it in sync here), so that's just a note on the side.
I think Peter is in a better position to give this the final greenlight, but it lgtm 👍🏼

**Review by peterbe — CHANGES_REQUESTED** [review]
As per discussed, we should make the unit tests for the lambda function also test this stuff at least once. E.g. `/sv-SE/docs/Foo?utm=bla`

File: deployer/aws-lambda/content-origin-request/server.test.js [hunk 1]
```diff
@@ -207,3 +207,34 @@ describe("redirect double-slash prefix URIs", () => {
...
+    expect(r.statusCode).toBe(302);
+    expect(r.headers["location"]).toBe(
+      "/en-US/docs/Web/HTML?retiredLocale=sv-SE"
+    );
+  });
+  it("should 302 redirect a retired locale (query string, improper locale)", async () => {
+    const r = await get("/BN/search?q=video");
+    expect(r.statusCode).toBe(302);
+    expect(r.headers["location"]).toBe(
+      "/en-US/search?retiredLocale=bn&q=video"
```
Comment (peterbe) [inline deployer/aws-lambda/content-origin-request/server.test.js:237]: Nice!

File: libs/fundamental-redirects/index.js [hunk 2]
```diff
@@ -92,6 +96,21 @@ const LOCALE_PATTERNS = [
...
+  redirect(
+    new RegExp(
+      `^(?<locale>${Array.from(RETIRED_LOCALES.keys()).join(
+        "|"
+      )})(/(?<suffix>.*)|$)`,
+      "i"
+    ),
+    ({ locale, suffix }) => {
+      const join = suffix && suffix.includes("?") ? "&" : "?";
+      return `/en-US/${
```
Comment (peterbe) [inline libs/fundamental-redirects/index.js:109]: I wonder, can you do 
```
const { DEFAULT_LOCALE } = require('../constants');
...
```
plus
```suggestion
      return `/${DEFAULT_LOCALE}/${
```

**Review by peterbe — APPROVED** [review]

File: libs/fundamental-redirects/index.js (hunk 2 above)
Comment (escattone) [reply libs/fundamental-redirects/index.js:109]: @peterbe Will do!

**Pull request merged**